Amicabilibus+RemoveDuplicates/src/Amicabilibus.java

```diff
@@ -1,35 +1,55 @@
+
 import java.util.ArrayList;
 
 public class Amicabilibus {
 	
-	public static final int upperbound = (int) Math.pow(10, 6);
+	public static final int upperbound = 901425; //(int) Math.pow(10, 6);
 	public static ArrayList<Integer> factorSums = new ArrayList<Integer>();
 	public static boolean[] checked = new boolean[upperbound+1];
 	
+//	public static int sumFactors(int n) {
+//		int sum = 1;
+//		for(int i=2;i <= (int)(Math.sqrt(n)); i++){
+//	    	if(n%i == 0){
+//	         	sum += i;
+//	           	if(i != n/i){sum += n/i;}
+//	       	}
+//	   	}
+//	    return sum;
+//	}
+	
 	public static int sumFactors(int n) {
-	    
+		if(n ==0) {
+			return 0;
+		}
 		int sum = 1;
-		for(int i=2;i <= (int)(Math.sqrt(n)); i++){
-	    	if(n%i == 0){
-	         	sum += i;
-	           	if(i != n/i){sum += n/i;}
-	       	}
-	   	}
-	    return sum;
+		int original = n;
+		for(int i=2; i*i<=n; i++) {
+			int product = 1;
+			while(n%i == 0) {
+				product=product*i+1;
+				n/=i;
+			}
+			sum*=product;
+		}
+		if(n>1) {sum*=1+n;}
+		sum -= original;
+		return sum;
 	}
 	
 	public static void main(String[] args) {
 		
 		for(int n=0; n<=upperbound; n++) {
 			factorSums.add(sumFactors(n));
 		}
+		System.out.println("done");
 		
 		for(int i=0; i<=upperbound; i++) {
 			int j = factorSums.get(i);
-			if(j <= upperbound && checked[i] == false && checked[j] == false && j!=i) {
+			if(j <= upperbound && checked[i] == false && j!=i) {
 				checked[i] = true; 
-				checked[j] = true;
 				if(i == factorSums.get(j)) {
+					checked[j] = true;
 					System.out.println("(" + i+ "," + factorSums.get(i) + ")");
```

Comment:
you don't need the `boolean[] checked` because you can just use the `ArrayList.contains()` to check the contains